### PR TITLE
test coverage: get rack instrumentation to 100%

### LINKED ISSUE #1427: Improve Rake test coverage
### Description
The `new_relic/agent/instrumentation/rake/instrumentation.rb` file has only 26% line coverage according to SimpleCov. Let's add tests to improve our coverage of this file to 80% or above. 

### Acceptance Criteria
* new_relic/agent/instrumentation/rake/instrumentation.rb test coverage according to SimpleCov is at least 80%

### Design Consideration/Limitations
* We don't need to add new instrumentation to achieve this goal

### Dependencies
None

### Additional context
Rake instrumentation is off by default for our customers, but it is still a commonly used tool.
Rake is not a core technology at this time.

### Estimates
S

### PR DESCRIPTION
**Pull request by fallwith**
raise code coverage for rack instrumentation to 100% line and 100% branch

### CODE REVIEW

File: test/multiverse/suites/padrino/Envfile [hunk 1]
```diff
@@ -5,8 +5,10 @@
 
 instrumentation_methods :chain, :prepend
 
+# TODO: sinatra 3's CALLERS_TO_IGNORE array is frozen, causes issues
 gemfile <<-RB
   gem 'activesupport', '~> 3'
+  gem 'sinatra', '< 3'
```
Comment (tannalynn) [inline test/multiverse/suites/padrino/Envfile:11]: I'm surprised this diff still shows this as new even though you merged the padrino PR already, any ideas why?

**Review by tannalynn — APPROVED** [review]

File: test/multiverse/suites/padrino/Envfile (hunk 1 above)
Comment (fallwith) [reply test/multiverse/suites/padrino/Envfile:11]: Yeah, this was the very first attempt at fix and it was slightly different. The final version that is already in dev has "sinatra" correctly listed in alphabetical order whereas it is not in order here. I'll make sure dev keeps the right version after everything is merged.

File: test/multiverse/suites/rack/rack_builder_test.rb [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+    def instance.deferred_dependency_check; end
+    def instance.check_for_late_instrumentation(obj); instance_variable_set(:@object_given, obj); end
+    arg = :the_arg
+    instance.with_deferred_dependency_detection { arg }
+    assert_equal instance.instance_variable_get(:@object_given), arg
+  end
+
+  def test_run_with_tracing
+    # TODO: OLD RUBIES - RUBY_VERSION < 2.3
+    skip 'Requires Ruby 2.3+' unless RUBY_VERSION >= '2.3.0'
```
Comment (kaylareopelle) [inline test/multiverse/suites/rack/rack_builder_test.rb:92]: Nice!

**Review by kaylareopelle — APPROVED** [review]
Thanks for improving our coverage on the other r*k tool!

File: test/multiverse/suites/rack/rack_builder_test.rb [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+    logger.expect(:info, nil, [/^We weren't/, /^To correct this/])
+
+    ::NewRelic::Agent::Instrumentation::RackHelpers.stub :middleware_instrumentation_enabled?, true do
+      ::NewRelic::Agent::Instrumentation::MiddlewareProxy.stub :needs_wrapping?, true do
+        ::NewRelic::Agent.stub :logger, logger do
+          instance.check_for_late_instrumentation(nil)
+        end
+      end
+    end
+    logger.verify
```
[the remark this answers is not in the document]
Comment (fallwith) [reply test/multiverse/suites/rack/rack_builder_test.rb:50]: With RSpec, you'd use either `allow` to permit something or `expect` to insist upon it.

With MiniTest, `expect` on its own behaves like an RSpec `allow` and then the use of `#verify` being called on the mock itself has it behave like an RSpec `expect`.

NOTE that for both frameworks, the existence of an allowed item triggers allow list behavior and no other method calls are permitted, so `allow` by itself can be valuable even if the allowed methods aren't invoked.

**Pull request merged**